src/components/Form/Form.md

```diff
@@ -354,6 +353,167 @@ const maxAge = 65
+  console.log("analyticsCustomEvent got called--field name: ", fieldName, "fieldValue: ", fieldValue, "--means no validation errors, so we'd likely call something like sendAnalyticsEvent(fieldName, fieldValue)")
+}
+
+;<Form
+  config={{
+    formName: 'Dynamic fields example form',
+    autocompleteOff: true,
+    formId: '1',
+    fields: {
+      buttonGroup: {
+        component: (props, options) => {
+          return (
+            <ButtonSelectGroup
+              {...props}
+              intialValue="often"
+            >
+              {options.map((x, i) => (
+                <ButtonSelectGroup.Option value={x.value} key={i}>
+                  {x.copy}
+                </ButtonSelectGroup.Option>
+              ))}
+            </ButtonSelectGroup>
+          )
+        },
+        validators: [validateIllegal],
+        validationSuccess: [analyticsCustomEvent],
+        labelCopy: 'Toggle dynamic fields',
+        options: [
+          { value: 'toggle-1', copy: 'Toggle Field 1' },
+          { value: 'toggle-2', copy: 'Toggle Field 2' },
+        ],
+      },
+      field1: {
+        component: (props, options) => {
+          return <TextInput
+            placeholder = "field 1 depends on choice 1--enter something for another dynamic field"
+            {...props}
+          />
+        },
+        validators: [(val) => val.length > 2 ? '2 chars max allowed' : ''],
+        optional: true,
+        labelCopy:
+          "optional with validation",
+        tid: 'field-1-text-input',
+      },
+      field2: {
+        component: (props, options) => {
+          return <TextInput
+            placeholder = "field 2 depends on choice 2"
+            {...props}
+          />
+        },
+        validators: [(val) => val.length > 2 ? '2 chars max allowed' : ''],
+        optional: true,
+        labelCopy:
+          "optional with validation",
+        tid: 'field-2-text-input',
+      },
+      field3: {
+        component: (props, options) => {
+          return <TextInput
+            placeholder = "field 3 depends on input in field 1"
+            {...props}
+          />
+        },
+        validators: [(val) => val.length > 2 ? '2 chars max allowed' : ''],
+        optional: true,
+        labelCopy:
+          "optional with validation",
+        tid: 'field-3-text-input',
+      },
+    },
+    onSubmit: async (formData) => {
+      if (count++ % 2 === 0) {
+        throw new Error("API Issue (Try again,, it alternates success & failure)")
+      } else {
+        alert(
+          'form submission successful with values:' +
+            JSON.stringify(formData)
+        )
+      }
+    },
+  }}
+>
+  {(api) => {
+    const {
+      field,
+      errors,
+      values,
+      getFormErrorMessage,
+      getFormIsValid,
+      getFormInteractedWith,
+    } = api
+    console.log('values', JSON.stringify(values, null, '     '))
+    console.log('errors', JSON.stringify(errors, null, '     '))
```

Comment:
this is wip but yeah this and the 'form interacted with.' message should probably be removed

Reply:
Yeah, I removed that this morning on the way in -- haven't pushed it up yet